ktor-client/ktor-client-curl/posix/src/io/ktor/client/engine/curl/internal/CurlMultiApiHandler.kt

```diff
@@ -58,13 +61,24 @@ internal class CurlMultiApiHandler : Closeable {
             option(CURLOPT_WRITEDATA, responseDataRef)
             option(CURLOPT_PRIVATE, responseDataRef)
             option(CURLOPT_ACCEPT_ENCODING, "")
+            request.connectTimeout?.let {
+                if (it != HttpTimeout.INFINITE_TIMEOUT_MS) {
```

Comment:
Sorry, didn't notice it during the previous iteration. It looks like we have the same issue as we had on iOS: the infinite timeout is not specified so it is [300 seconds by default](https://curl.haxx.se/libcurl/c/CURLOPT_CONNECTTIMEOUT_MS.html).

Reply:
Thanks, I fixed it.